userstorage/backend.py

```diff
@@ -92,12 +96,21 @@ def teardown(self):
         Subclass should override if needed.
         """
 
+    def missing_handler(self, msg):
+        """
+        Should be called when entering the context without having created
+        the associated storage.
+
+        Subclass can override if needed.
+        """
+        raise Missing(msg)
+
     def __enter__(self):
         """
         Setup the current storage and return it.
         """
         if not self.exists():
-            pytest.xfail(f"backend {self} not available")
+            self.missing_handler(f"backend {self} not available")
```

Comment:
Looks good, but the handler should configurable. Otherwise users will need to wrap all the tests or fixtures using this to catch the error.

You don't want that user will subclass all the backends to handle this.

Reply:
Wanted to keep it simple, but is true that it would force users to keep using a custom Backend (or similar) to catch the exception.

Let's see if I can manage to make it configurable properly.